packages/client/test/rpc/engine/newPayloadV1.spec.ts

```diff
@@ -40,16 +45,24 @@ tape(`${method}: call with invalid block hash without 0x`, async (t) => {
 tape(`${method}: call with invalid hex string as block hash`, async (t) => {
   const { server } = baseSetup({ engine: true, includeVM: true })
 
-  const req = params(method, [
-    {
-      ...validTestVector,
-      blockHash: '0xinvalid',
-    },
-  ])
+  const blockDataWithInvalidBlockHash = [{ ...blockData, blockHash: '0x-invalid-block-hash' }]
+  const req = params(method, blockDataWithInvalidBlockHash)
   const expectRes = checkError(
     t,
     INVALID_PARAMS,
     "invalid argument 0 for key 'blockHash': invalid block hash"
   )
   await baseRequest(t, server, req, 200, expectRes)
 })
+
+// tape(`${method}: call with valid data`, async (t) => {
+//   const { server } = await setupChain(genesisJSON, 'post-merge', { engine: true })
+//
+//   const req = params(method, [blockData])
+//
+//   const expectRes = (res: any) => {
+//     console.log(res)
+//     // t.equal(res.body.result, 'SYNCING')
```

Comment:
When running this method, I am getting this error 
> Error verifying block while running: Error: BlockHeader.cliqueIsEpochTransition() call only supported for clique PoA networks (block header number=1 hash=0x3559e851470f6e7bbed1db474980683e8c315bfce99b2a6ef47c057c04de7858 hf=merge baseFeePerGas=7)'


It's being triggered when [`putBlock`](https://github.com/ethereumjs/ethereumjs-monorepo/blob/test/engine-api/packages/client/lib/rpc/modules/engine.ts#L366) method is executed. After debugging for some time, I noticed that there's a mismatch between `this` and `header` variables 
```ts
this._common.consensusAlgorithm() -> 'clique'
header._common.consensusAlgorithm() -> 'casper'
```


https://github.com/ethereumjs/ethereumjs-monorepo/blob/c9d5fe97e2e4383bdd1d06a61b8c0aa80ced07ab/packages/blockchain/src/index.ts#L951
I think this is related (again) to the way we are instantiating the client

Will try to keep debugging later, but if someone has any idea I am all ears :ear:

Reply:
hm well for post-merge consensusAlgorithm() shouldn't return clique, since TTD is 0 it should always return casper. i checked and setupChain does call common.setHardforkByBlockNumber(0) so that seems right. which this._common.consensusAlgorithm() is returning clique? I'll have to check out the code and see what's going on.

Reply:
After doing that, I noticed that the common `hardfork` variable is still `london`, not `merge`, and that's the reason why we are getting this missmatch

Reply:
I found the problem! :-D Indeed we needed to pass the `TD` param from the Geth Config. I went deep and tried to understand why do we need this, and if I understood correctly, I can say that the reason of this mismatch behavior between the integration tests with lodestar and w/this unit tests, is that, since we are mocking the client, we need to explicitly set the the `hardfork` based on the block number; but when running the process from the CLI, it does this when the `Synchronizer` opens, which is this line https://github.com/ethereumjs/ethereumjs-monorepo/blob/merge-kiln/packages/client/lib/sync/fullsync.ts#L61

Would that make sense to you? @ryanio

Reply:
correct, nice find!